### PR TITLE
MODCLUSTER-738 Remove service loader mechanism which was in place to …

### PR DESCRIPTION
**Pull request by rhusar**
…support code sharing between Tomcat modules

Resolves
https://issues.redhat.com/browse/MODCLUSTER-738

### CODE REVIEW

File: container/tomcat-10.0/src/main/java/org/jboss/modcluster/container/tomcat/TomcatEventHandlerAdapter.java [hunk 1]
```diff
@@ -49,7 +48,7 @@
 
     protected final ContainerEventHandler eventHandler;
     protected final ServerProvider serverProvider;
-    protected final TomcatFactory factory;
+    protected TomcatRegistry registry;
```
Comment (rhusar) [inline container/tomcat-10.0/src/main/java/org/jboss/modcluster/container/tomcat/TomcatEventHandlerAdapter.java:51]: final

Comment (pferraro) [inline container/tomcat-10.0/src/main/java/org/jboss/modcluster/container/tomcat/TomcatEventHandlerAdapter.java:51]: Can this remain final?

File: container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java [hunk 2]
```diff
@@ -45,16 +47,20 @@
...
+    protected final TomcatRegistry registry = mock(TomcatRegistry.class);
     protected final ContainerEventHandler eventHandler = mock(ContainerEventHandler.class);
     protected final LifecycleServer server = mock(LifecycleServer.class);
-    protected final TomcatFactory factory = mock(TomcatFactory.class);
+    protected final Object factory = mock(Object.class);
     protected final ServerProvider provider = mock(ServerProvider.class);
 
-    protected TomcatEventHandler createEventHandler(ContainerEventHandler eventHandler, ServerProvider provider, TomcatFactory factory) {
-        return new TomcatEventHandlerAdapter(eventHandler, provider, factory);
+    protected TomcatEventHandler createEventHandler(ContainerEventHandler eventHandler, ServerProvider provider, Object ignore) {
```
Comment (pferraro) [inline container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java:62]: Did you mean to leave this "ignore" parameter?

Comment (rhusar) [inline container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java:62]: remove ignore

File: container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java [hunk 3]
```diff
@@ -217,13 +220,13 @@ public void undeployWebApp() throws Exception {
...
         when(service.getServer()).thenReturn(server);
-        when(this.factory.createContext(context)).thenReturn(catalinaContext);
+//        when(this.factory.createContext(context)).thenReturn(catalinaContext);
 
         handler.containerEvent(event);
 
         verify(context).removeLifecycleListener(handler);
         verify(context).removePropertyChangeListener(handler);
-        verify(this.eventHandler).remove(same(catalinaContext));
+//        verify(this.eventHandler).remove(same(catalinaContext));
```
Comment (rhusar) [inline container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java:229]: reenable

Comment (rhusar) [reply container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java]: Done.

File: container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java (hunk 2 above)
Comment (rhusar) [reply container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java]: Done.

Comment (rhusar) [reply container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java]: Done.

File: container/tomcat-10.0/src/main/java/org/jboss/modcluster/container/tomcat/TomcatEventHandlerAdapter.java (hunk 1 above)
Comment (rhusar) [reply container/tomcat-10.0/src/main/java/org/jboss/modcluster/container/tomcat/TomcatEventHandlerAdapter.java]: Done.

Comment (rhusar) [reply container/tomcat-10.0/src/main/java/org/jboss/modcluster/container/tomcat/TomcatEventHandlerAdapter.java]: Done.

**Review by pferraro — APPROVED** [review]

**Comment by rhusar** [comment]
Thanks @pferraro – was running a couple of tests and thing seem to be working alright.

**Pull request merged**